src/components/bookunits.js

```diff
@@ -1,30 +1,33 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
 
-const BookUnits = () => (
-  <section id="shelf-book">
-    <div id="nav-1/3">
-      <p>category</p>
-      <h3>book title</h3>
-      <p>author</p>
-      <div>
-        <p>comments</p>
-        <a href="/">remove</a>
-        <p>edit</p>
-      </div>
-    </div>
-    <div id="nav-2/3">
-      <img alt="login" />
-      <div>
-        <p>%</p>
-        <p>completed</p>
+const BookUnits = ({ id, title, author }) => {
+  const dispatch = useDispatch();
+
+  return (
+    <div id={id}>
+      <div key={id}>
+        <div>{title}</div>
+        <div>{author}</div>
+        <button
+          type="button"
+          onClick={(e) => {
+            dispatch(removeBook(e.target.id));
+          }}
+        >
```

Comment:
- [ ] `removeBook` is not removing a book as currently implemented. Looking at the screenshot below tells us that there is no change in the previous state and the next state after dispatching the `REMOVE_BOOK` action.

  ![image](https://user-images.githubusercontent.com/45002654/151085571-e10b456a-2d4c-464b-9f83-16800062d8c6.png)
  
  Why is that? Look at the `id` value passed to the action, it's an empty string.
  
  ### The first step to solving this:
  We are extracting `id` from `e.target` which refers to this button but the button itself has no `id` property. That problem number one. Please add the `id` property as below:
  ```ruby
  <button
    id={id}
    type="button"
    onClick={(e) => {
      dispatch(removeBook(e.target.id));
    }}
  >
  ```

Reply:
Thank you so much for the good review!

Reply:
You are welcome @fabianofrank

Reply:
This issue is well resolved! ✔️